sympy/physics/mechanics/_pathway.py

```diff
@@ -0,0 +1,253 @@
+    >>> linear_pathway = LinearPathway(attachments)
+    >>> linear_pathway
+    LinearPathway(attachments=(pO, pI))
+
+    The pathway create above isn't very interesting without the positions and
+    velocities of its attachment points being described. Without this its not
+    possible to describe how the pathway moves, i.e. its length or its
+    shortening velocity.
+
+    >>> from sympy.physics.mechanics import ReferenceFrame
+    >>> from sympy.physics.vector import dynamicsymbols
+    >>> N = ReferenceFrame('N')
+    >>> q = dynamicsymbols('q')
+    >>> attachments[-1].set_pos(attachments[0], q * N.x)
+    >>> attachments[-1].pos_from(attachments[0])
+    q(t)*N.x
+
+    A pathway's length can be accessed via its ``length`` attribute.
+
+    >>> linear_pathway.length
+    sqrt(q(t)**2)
+
+    Note how what appears to be an overly-complex expression is returned. This
+    is actually required as it ensures that a pathway's length is always
+    positive.
+
+    A pathway's shortening velocity can be accessed similarly via its
+    ``shortening_velocity`` attribute.
+
+    >>> linear_pathway.shortening_velocity
+    -q(t)*Derivative(q(t), t)/sqrt(q(t)**2)
+
+    Parameters
+    ==========
+
+    attachments : tuple[Point, Point]
+        The pair of ``Point`` objects between which the linear pathway spans.
+
+    """
+
+    def __init__(
+        self,
+        attachments: tuple[Point, ...],
+    ) -> None:
+        """Initializer for ``LinearPathway``.
+
+        Parameters
+        ==========
+
+        attachments : tuple[Point, Point]
+            The pair of ``Point`` objects between which the linear pathway
+            spans.
+
+        """
+        super().__init__(attachments)
+
+    def _true_length(self) -> Expr:
+        """Exact analytical expression for the pathway's length."""
+        length = self.attachments[-1].pos_from(self.attachments[0]).magnitude()
+        return length
+
+    def _true_shortening_velocity(self) -> Expr:
+        """Exact analytical expression for the pathway's shortening velocity."""
+        relative_position = self.attachments[-1].pos_from(self.attachments[0])
+        if not relative_position:
+            return S.Zero
+        t = dynamicsymbols._t  # type: ignore
+        # A reference frame is needed to differentiate ``relative_position`` to
+        # ``relative_velocity`` so choose the first ``ReferenceFrame`` that
+        # ``relative_position`` is defined using.
+        frame = relative_position.args[0][1]
+        relative_velocity = relative_position.diff(t, frame)
+        shortening_velocity = -relative_velocity.dot(relative_position.normalize())
+        return shortening_velocity
+
+    def compute_loads(self, force: Symbol) -> list[tuple[Point, Vector]]:
+        """Loads required by the equations of motion method classes.
+
+        Explanation
+        ===========
+
+        ``KanesMethod`` requires a ``list[tuple[Point, Vector]]`` to be passed
```

Comment:
```
``list[tuple[Point, Vector]]``
```
Should be:
```
``list[tuple[Point, Vector], ...]``
```

Reply:
For type hints `list[tuple[Point, Vector]]` means that it's a list of `(Point, Vector)` and that that list can be of any length, i.e. the ellipsis isn't needed. But given that this is a description using the type hinting syntax may not be the most clear to the reader. I'll update.

Reply:
Addressed in https://github.com/sympy/sympy/pull/24914/commits/df2fe5e7fa4c50032bf1e8a8ef38ebb2d2a6a63f.